payjoin/src/receive/error.rs

```diff
@@ -41,6 +41,21 @@ impl error::Error for Error {
     }
 }
 
+#[derive(Debug)]
+/// Error arising during the persistence of the receiver
+pub enum PersistanceError {
+    PersistError(ImplementationError),
+}
+impl std::fmt::Display for PersistanceError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PersistanceError::PersistError(e) => write!(f, "{}", e),
+        }
+    }
+}
+
+impl std::error::Error for PersistanceError {}
+
```

Comment:
What does this give us beyond the existing ImplementationError type? The display function doesn't even add context. I'd rather just use `ImplementationError` if the answer is indeed "nothing."

Reply:
I was following the pattern set by `ReplayableError`. `?` Doesnt work for trait object (`dyn StdError + Send + Sync`) because they are not sized. I could just re-use `ReplayableError` but that seemed like broke the seperation of concern for that error enum. 
```
The replyable error type for the payjoin receiver, representing failures need to be
/// returned to the sender.
```
 Since persistance is never returned to the sender I opted in for a seperate error variant

Reply:
Alternatively we could Box it.  `Box<ImplementationError>`. This would be my prefrence

Reply:
I'm not sure what "Doesn't work for trait object because they are not sized." means here. If not returning `Result<Receiver, ImplementationError>` why not follow the pattern used elsewhere in this module and return `Result<Receiver, ReplyableError>`?

Reply:
>I'm not sure what "Doesn't work for trait object because they are not sized." means here

`ImplementationError` is a typedef for `dyn StdError + Send + Sync` which is not sized meaning you cannot use `?`.

>why not follow the pattern used elsewhere in this module and return Result<Receiver, ReplyableError>?

 `ReplyableError` describes an error that is only being returned to the sender. Happy to use `ReplayableError` even though this sender will never know about this error just seemed more intuitive to introduce a new error variant

Reply:
Done